src/test/java/org/galatea/starter/entrypoint/IexRestControllerTest.java

```diff
@@ -74,11 +75,94 @@ public void testGetLastTradedPrice() throws Exception {
+        .andExpect(jsonPath("$[0].volume").value(17362305))
+        .andExpect(jsonPath("$[0].date", is("2021-05-10")))
+        .andReturn();
+  }
+
+  @Test
+  public void testGetHistoricalPriceSymbolDate() throws Exception {
+
+    MvcResult result = this.mvc.perform(
+        MockMvcRequestBuilders
+            .get("/iex/historicalPrice?symbol=twtr&date=20210405")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].close").value(new BigDecimal("64.24")))
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("64.34")))
+        .andExpect(jsonPath("$[0].low").value(new BigDecimal("61.81")))
+        .andExpect(jsonPath("$[0].open").value(new BigDecimal("64.14")))
+        .andExpect(jsonPath("$[0].symbol", is("TWTR")))
+        .andExpect(jsonPath("$[0].volume").value(15502130))
+        .andExpect(jsonPath("$[0].date", is("2021-04-05")))
+        .andReturn();
+  }
+
+  @Test
+  public void testGetHistoricalPriceSymbolRange() throws Exception {
+
+    MvcResult result = this.mvc.perform(
+        MockMvcRequestBuilders
+            .get("/iex/historicalPrice?symbol=twtr&range=3d")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].close").value(new BigDecimal("57.01")))
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("58.67")))
+        .andExpect(jsonPath("$[0].low").value(new BigDecimal("55.83")))
+        .andExpect(jsonPath("$[0].open").value(new BigDecimal("56.96")))
+        .andExpect(jsonPath("$[0].symbol", is("TWTR")))
+        .andExpect(jsonPath("$[0].volume").value(21985673))
+        .andExpect(jsonPath("$[0].date", is("2021-06-03")))
+        .andReturn();
+  }
+
+  @Test
+  public void testGetHistoricalPriceSymbolRangeDate() throws Exception {
+
+    MvcResult result = this.mvc.perform(
+        MockMvcRequestBuilders
+            .get("/iex/historicalPrice?symbol=twtr&range=1m&date=20210210")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].close").value(new BigDecimal("67.77")))
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("69.25")))
+        .andExpect(jsonPath("$[0].low").value(new BigDecimal("63.2")))
+        .andExpect(jsonPath("$[0].open").value(new BigDecimal("65.8")))
+        .andExpect(jsonPath("$[0].symbol", is("TWTR")))
+        .andExpect(jsonPath("$[0].volume").value(73649183))
+        .andExpect(jsonPath("$[0].date", is("2021-02-10")))
+        .andReturn();
+  }
+
+  @Test
+  public void testGetHistoricalPriceEmpty() throws Exception {
+
+    MvcResult result = this.mvc.perform(
+        MockMvcRequestBuilders
+            .get("/iex/historicalPrice?symbol=")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isBadRequest())
+        .andReturn();
+  }
```

Comment:
Please add one more test to validate the case if the IEX server is down or fails.